presto-spi/src/main/java/com/facebook/presto/spi/type/SqlTime.java

```diff
@@ -67,6 +84,12 @@ public boolean equals(Object obj)
     @Override
     public String toString()
     {
-        return Instant.ofEpochMilli(millisUtc).atZone(ZoneId.of(sessionTimeZoneKey.getId())).format(formatter);
+        if (isDeprecatedTimestamp()) {
+            return Instant.ofEpochMilli(millisUtc).atZone(ZoneId.of(sessionTimeZoneKey.get().getId())).format(formatter);
+        }
+        else {
+            // FIXME this should be exact equivalent of TIMESTAMP to VARCHAR cast
+            return Instant.ofEpochMilli(millisUtc).atZone(ZoneId.of(UTC_KEY.getId())).format(formatter);
```

Comment:
What about leap seconds. Isn't modulo approach broken if we take them into consideration?